logging/telemetryhook.go

```diff
@@ -146,6 +146,9 @@ func (hook *asyncTelemetryHook) waitForEventAndReady() bool {
 
 // Fire is required to implement logrus hook interface
 func (hook *asyncTelemetryHook) Fire(entry *logrus.Entry) error {
+	if _, ok := entry.Data["TelemetryError"]; ok {
+		return nil
+	}
```

Comment:
Is this a better location than [`telemetryFilteredhook`](https://github.com/algorand/go-algorand/blob/master/logging/telemetryFilteredHook.go#L52), as to avoid it being added to the async queue in the first place? `telemetryFilteredhook` would be a better location if that concern isn't true.